utils/sample_qc.py

```diff
@@ -364,91 +364,109 @@ def infer_sex(
-    yy_y_ploidy_cutoff: float = 1.3,
-    xxx_x_ploidy_cutoff:float = 2.5,
-    f_stat_female_cutoff: float = -0.2,
-    f_stat_male_cutoff: float = 0.2
+    ht: hl.Table,
 ) -> hl.expr.StructExpression:
-    # TODO: Automate cutoffs
-    # TODO: Provide alternatives in case of missing annotations (e.g. no chr_y_ploidy)
     """
     Creates a struct with the following annotations:
-    - karyoptype (str)
-    - sex (str), which can be either 'male', 'female' or missing
+    - sex_karyoptype (str)
+    - sex (str), which can be either 'male', 'female', 'ambiguous_sex', or 'sex_aneuploidy'
     - is_female (missing for karyotypes that aren't either 'XX' or 'XY')
 
-    :param NumericExpression chr_x_ploidy: chrom X ploidy (or relative ploidy)
-    :param NumericExpression  chr_y_ploidy: chrom Y ploidy (or relative ploidy)
-    :param NumericExpression f_stat: chrom X F-stat
-    :param Tuple[float, float] xx_x_ploidy_cutoffs: Boundaries around the chom X ploidy for females
-    :param Tuple[float, float] xy_x_ploidy_cutoffs: Boundaries around the chom X ploidy for males
-    :param float xx_y_ploidy_cutoff: Max y ploidy for females
-    :param Tuple[float, float] xy_y_ploidy_cutoffs: Boundaries around the chom Y ploidy for males
-    :param float yy_y_ploidy_cutoff: Min chrom Y ploidy for YY
-    :param float xxx_x_ploidy_cutoff: Min chrom X ploidy for XXX
-    :param float f_stat_female_cutoff: Min f-stat for females
-    :param flat f_stat_male_cutoff: Max f-stat for males
-    :return: Struct expression with sex anotations
+    :param Table ht: Table with samples, their imputed sex, and their sex chromosome ploidies
     :rtype: StructExpression
     """
+    def get_ploidy_cutoffs(ht: hl.Table, sex: str) -> dict:
+        """
+        Gets chromosome X and Y ploidy cutoffs for XY and XX samples
+
+        :param Table ht: Table with samples, their imputed sex, and their sex chromosome ploidies
+        :param str sex: 'XY' or 'XX'
+        :return: Dictionary of ploidy cutoffs; key: sex chrom, values: [lower cut, upper cut, outlier (aneuploidy) cut]
+        :rtype: dict
+        """
+        if sex == "XY":
+            ht = ht.filter((hl.is_defined(ht.is_female)) & (~ht.is_female))
+        else:
+            ht = ht.filter(ht.is_female)
+
+        # Get mean/stdev for chrX/Y ploidies
+        X_stats = ht.aggregate(hl.agg.stats(ht.chrX_ploidy))
+        Y_stats = ht.aggregate(hl.agg.stats(ht.chrY_ploidy))
+
+        # Set cutoffs -- keep cutoffs permissive to account for loss of sex chromosomes
+        # NOTE: keep lower bound of Y particularly permissive (0.05) to save males with somatic loss of Y 
+        cutoffs = {}
+        cutoffs["X"] = [X_stats.mean - (5 * X_stats.stdev), X_stats.mean + (5 * X_stats.stdev), X_stats.mean + (6 * X_stats.stdev)]
+        cutoffs["Y"] = [min(0.05, Y_stats.mean - (6 * Y_stats.stdev)), Y_stats.mean + (5 * Y_stats.stdev), Y_stats.mean + (6 * Y_stats.stdev)]
```

Comment:
I don't think we should have a `min(0.05...` here --  this makes some assumption about 0.05 being a reasonable cutoff. My preference would be something like 2*n_SDs (or more if necessary).

Reply:
I also think that we possibly  need to allow for somatic loss of X in females BTW